File: java/eshop-cqrs-ddd-microservices-excercise/common/src/main/java/com/marekturis/common/domain/event/EventBase.java
package com.marekturis.common.domain.event;

import java.util.Date;
import java.util.UUID;

/**
 * @author Marek Turis
 */
public class EventBase implements Event {

	private String identity;
	private Date occuredOn;

	public EventBase() {
		occuredOn = new Date();
		identity = UUID.randomUUID().toString();
	}

	@Override
	public String identity() {
		return identity;
	}

	@Override
	public Date occuredOn() {
		return occuredOn;
	}

	@Override
	public String name() {
		return getClass().getName();
	}
}

File: dotnet/eshop-cqrs-ddd-microservices-excercise/Com.Marekturis.Common/Domain/Event/EventBase.cs
using System;

namespace Com.Marekturis.Common.Domain.Event
{
    public class EventBase : Event
    {
        public DateTime OccuredOn { get; private set; }
        public string Name { get; private set; }

        public EventBase()
        {
            OccuredOn = DateTime.Now;
            Name = GetType().Name;
        }
    }
}